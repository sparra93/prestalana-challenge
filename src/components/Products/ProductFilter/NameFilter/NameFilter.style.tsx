import styled from 'styled-components';

export const Input = styled.input`
  padding: 8px;
  font-size: 14px;
  margin-bottom: 20px;
  display: block;
  width: auto;
`;
