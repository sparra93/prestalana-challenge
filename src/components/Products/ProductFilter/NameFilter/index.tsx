import type { ReactElement } from 'react';
import React from 'react';

import Panel from '@/components/Panel';
import { Input } from '@/components/Products/ProductFilter/NameFilter/NameFilter.style';

interface NameFilterProps {
  onChange: (name: string) => void;
}

const NameFilter: React.FC<NameFilterProps> = ({ onChange }): ReactElement => {
  const handleNameChange = (
    event: React.ChangeEvent<HTMLInputElement>,
  ): void => {
    onChange(event.target.value);
  };

  return (
    <Panel title="Nombre">
      <Input
        type="text"
        onChange={handleNameChange}
        placeholder="Product name"
      />
    </Panel>
  );
};

export default NameFilter;
