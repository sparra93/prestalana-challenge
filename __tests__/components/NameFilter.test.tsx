import { fireEvent, screen } from '@testing-library/react';
import React from 'react';

import NameFilter from '@/components/Products/ProductFilter/NameFilter';

describe('NameFilter Component', () => {
  const onChangeInput = jest.fn();

  it('should display NameFilter information correctly', () => {
    renderWithProviders(<NameFilter onChange={onChangeInput} />);

    const input = screen.getByPlaceholderText(`Product name`);
    expect(input).toBeInTheDocument();
  });

  it('should do type and call the function', () => {
    renderWithProviders(<NameFilter onChange={onChangeInput} />);

    const input = screen.getByPlaceholderText(`Product name`);
    fireEvent.change(input, { target: { value: 'test' } });
    expect(input).toBeInTheDocument();
    expect(onChangeInput).toHaveBeenCalledWith('test');
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProviders(
      <NameFilter onChange={onChangeInput} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
