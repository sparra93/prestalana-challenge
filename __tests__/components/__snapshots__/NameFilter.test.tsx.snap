// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameFilter Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-c77c5486-0 gRkaAU"
    draggable="false"
  >
    <div
      class="sc-c77c5486-1 iDpoqc"
    >
      <span>
        Nombre
      </span>
    </div>
    <div
      class="sc-c77c5486-2 eTCBvC"
    >
      <input
        class="sc-93af5939-0 bHotKZ"
        placeholder="Product name"
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
